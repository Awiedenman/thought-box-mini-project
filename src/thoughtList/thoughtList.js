import React from 'react';
import { ThoughtCard } from '../thoughtCard/thoughtCard';

export const ThoughtList = ( {thoughtList} ) => {
  
  const createThoughts = thoughtList.map(( thought, index) =>{
    return < ThoughtCard 
              { ...thought }
              key={ index }
              />
  })

  return (
    <div>
   { createThoughts }
    </div>
  );
}
