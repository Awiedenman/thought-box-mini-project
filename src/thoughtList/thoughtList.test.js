import React from 'react';
import { shallow, mount } from 'enzyme';

import { ThoughtList } from './thoughtList.js';

describe('ThoughtList', () => {

  it('should match the snapshot with no thoughts', () => {
    const mockState = [];
    const wrapper = shallow(
      <ThoughtList
        thoughtList={mockState}
      />)
    
      expect(wrapper).toMatchSnapshot()
  });

  it('should match the snapshot with thoughts', () => {
    const mockState = [ 
      {body: "c", id: 0, title: "c"}, 
      {body: "d", id: 1, title: "d"}  
    ];
    const wrapper = shallow(
      <ThoughtList  
        thoughtList={mockState}
      />)

      expect(wrapper).toMatchSnapshot()
  });

});
