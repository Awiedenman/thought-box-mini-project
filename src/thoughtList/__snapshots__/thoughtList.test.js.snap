// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThoughtList should match the snapshot with no thoughts 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    thoughtList={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ThoughtList should match the snapshot with thoughts 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    thoughtList={
        Array [
            Object {
              "body": "c",
              "id": 0,
              "title": "c",
            },
            Object {
              "body": "d",
              "id": 1,
              "title": "d",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          body="c"
          id={0}
          title="c"
/>,
        <Unknown
          body="d"
          id={1}
          title="d"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "body": "c",
          "id": 0,
          "title": "c",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "body": "d",
          "id": 1,
          "title": "d",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            body="c"
            id={0}
            title="c"
/>,
          <Unknown
            body="d"
            id={1}
            title="d"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "body": "c",
            "id": 0,
            "title": "c",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "body": "d",
            "id": 1,
            "title": "d",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
